### PR TITLE
Add Hair Dye

### PR DESCRIPTION
**Pull request by MNG-cataclysm**
#### Summary
Content "Add hair dye."

#### Purpose of change
Add hair dyes, with a way to use them to change hair color. Part of an ongoing cosmetic project.

#### Describe the solution
Using EOCs, add a hair dye kit that players can find in barber shops, allowing them to apply a hair dye of their choice. This includes all of the current colors of hair, black, brown, blond, blue, white, gray, pink, green, and red.

#### Describe alternatives you've considered
Not doing this.

#### Testing
Walked around barber shops and found the hair dye kits and bottles of dye. Applied each as such, testing with tilesets. Hair color successfully changed with no hiccups.

#### Additional context
Currently, hair dye is permanent and doesn't wear off, unless your hair grows out with Natural Hair Growth. I'd like to add dye to it's own independent timer that wears off after 6-8 weeks.

After everything else is done, #71919 should probably be considered for artificial coloring purposes.

### CODE REVIEW

**Comment by fairyarmadillo** [comment]
You have no idea how much I have tried to get this to work 😆 thank you for putting it together.

File: data/json/effects_on_condition/npc_eocs/appearance_eocs.json [hunk 1]
```diff
@@ -232,6 +232,137 @@
...
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    ]
+  },
```
Comment (Inglonias) [inline data/json/effects_on_condition/npc_eocs/appearance_eocs.json:320]: holy stack of closing braces, batman.

Comment (MNG-cataclysm) [reply data/json/effects_on_condition/npc_eocs/appearance_eocs.json:320]: I don’t know how to fix that, but I’ll look into it.

Comment (Inglonias) [reply data/json/effects_on_condition/npc_eocs/appearance_eocs.json:320]: You don't have to fix it. I just think it's funny

**Comment by Maleclypse** [comment]
Thank you this has been cool watching you go after hair :)

**Comment by MNG-cataclysm** [comment]
> Thank you this has been cool watching you go after hair :)

You're quite welcome, it's been a fun little project.

**Pull request merged**